tests/drivers/flash/api/page_layout/src/main.c

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2020-2022 Nordic Semiconductor ASA
+ * Copyright (c) 2024 Laczen
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <zephyr/kernel.h>
+#include <zephyr/ztest.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/flash.h>
+
+#define TEST_DEVICE flash_sim0
+#if !DT_NODE_EXISTS(DT_NODELABEL(TEST_DEVICE))
+#undef TEST_DEVICE
+#define TEST_DEVICE flash0
```

Comment:
Also, the test device I would expect to be the flash controller, not the flash area, maybe rename `TEST_DEVICE` to `TEST_AREA` or `TEST_FLASH_AREA`?